plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,371 @@
+		t.Period = config.Duration(5 * time.Minute)
+	}
+	if t.Delay <= 0 {
+		t.Delay = config.Duration(0 * time.Minute)
+	}
+
+	if len(t.Accounts) == 0 {
+		return fmt.Errorf("account is empty")
+	}
+
+	t.discoverTool = NewDiscoverTool(t.Log)
+
+	// create account credential
+	for i := range t.Accounts {
+		if t.Accounts[i].SecretID == "" {
+			return fmt.Errorf("secret_id is empty")
+		}
+		if t.Accounts[i].SecretKey == "" {
+			return fmt.Errorf("secret_key is empty")
+		}
+
+		t.Accounts[i].crs = common.NewCredential(
+			t.Accounts[i].SecretID,
+			t.Accounts[i].SecretKey,
+		)
+		if t.Accounts[i].Name == "" {
+			t.Accounts[i].Name = fmt.Sprintf("account_%v", i)
+		}
+		// check if namespace supports auto discovery
+		for _, namespace := range t.Accounts[i].Namespaces {
+			if namespace.Name == "" {
+				return fmt.Errorf("namespace is empty")
+			}
+			for _, region := range namespace.Regions {
+				if region.RegionName == "" {
+					return fmt.Errorf("region is empty")
+				}
+				_, ok := t.discoverTool.registry[namespace.Name]
+				if len(region.Instances) == 0 && !ok {
+					return fmt.Errorf(
+						"unsupported namespace %s for discovering instances, please specify instances and dimensions",
+						namespace.Name,
+					)
+				}
+			}
+		}
+	}
+
+	// start discovery
+	go t.discoverTool.Discover(t.Accounts, t.DiscoveryInterval, t.Endpoint)
+	t.discoverTool.DiscoverMetrics()
+	for i := range t.Accounts {
+		for j := range t.Accounts[i].Namespaces {
+			// use discovered metrics if not specified
+			if len(t.Accounts[i].Namespaces[j].Metrics) == 0 {
+				metrics, ok := t.discoverTool.DiscoveredMetrics[t.Accounts[i].Namespaces[j].Name]
+				if !ok {
+					return fmt.Errorf("unsupported namespace %s for discovering metrics, please specify metrics",
+						t.Accounts[i].Namespaces[j].Name)
+				}
+				t.Accounts[i].Namespaces[j].Metrics = metrics
+			}
+		}
+	}
+
+	t.client = &cloudmonitorClient{Accounts: t.Accounts}
+
+	return nil
+
+}
+
+func (t *TencentCloudCM) updateWindow(relativeTo time.Time) {
+
+	windowEnd := relativeTo.Add(-time.Duration(t.Delay))
+
+	// starting point is two times the aggregation period to make sure all points are covered
+	t.windowStart = windowEnd.Add(-time.Duration(t.Period) * 2)
+
+	t.windowEnd = windowEnd
+}
+
+func (t *TencentCloudCM) Gather(acc telegraf.Accumulator) error {
+	t.updateWindow(time.Now())
+
+	metricObjects := t.client.GetMetricObjects(*t)
+
+	lmtr := limiter.NewRateLimiter(t.RateLimit, time.Second)
+	defer lmtr.Stop()
+
+	wg := sync.WaitGroup{}
+	rLock := sync.Mutex{}
+	results := []monitor.GetMonitorDataResponse{}
+
+	// requestIDMap contains request ID and metric objects for later aggregation
+	requestIDMap := map[string]metricObject{}
+	for _, obj := range metricObjects {
+
+		wg.Add(1)
+		<-lmtr.C
+		go func(m metricObject) {
+			defer wg.Done()
+
+			for {
+
+				client, err := t.client.NewClient(m.Region, m.Account.crs, *t)
+				if err != nil {
+					acc.AddError(err)
+					break
```

Comment:
You might want to use `return` here to make clear you do not expect to do any further processing here.

Reply:
fixed in 103f841